pkg/apis/submarinerconfig/v1alpha1/types.go

```diff
@@ -195,6 +195,79 @@ type ManagedClusterInfo struct {
 	InfraID string `json:"infraId,omitempty"`
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Namespaced"
+
+// SubmarinerDiagnoseConfig represents the configuration for SubmarinerDiagnose, the submariner-addon will use it
+// to run SubmarinerDiagnose.
+type SubmarinerDiagnoseConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the configuration of the Submariner
+	Spec SubmarinerDiagnoseSpec `json:"spec"`
+
+	// Status represents the current status of submariner configuration
+	// +optional
+	Status SubmarinerDiagnoseStatus `json:"status,omitempty"`
+}
+
+// SubmarinerDiagnoseSpec defines the desired configuration to run SubmarinerDiagnose.
+type SubmarinerDiagnoseSpec struct {
+	All             bool            `json:"all,omitempty"`
+	CNI             bool            `json:"CNI,omitempty"`
+	Connections     bool            `json:"connections,omitempty"`
+	Deployment      bool            `json:"deployment,omitempty"`
```

Comment:
Where are the results of these two captured?

Reply:
ManagedClusterAddon Status field already captures Deployment status. Not 100% sure if it is set only at initial phase or it can be updated if something goes wrong later. We might need to add something here for Deployment status. Can always add it later through a Issue/PR if needed.

Connections status is captured as part of Gateways object and we're going to use that to display ConnectionStatus as detailed in https://github.com/stolostron/backlog/issues/19294

Reply:
> ManagedClusterAddon Status field already captures Deployment status. Not 100% sure if it is set only at initial phase or it can be updated if something goes wrong later. We might need to add something here for Deployment status. Can always add it later through a Issue/PR if needed.
> 
> Connections status is captured as part of Gateways object and we're going to use that to display ConnectionStatus as detailed in [stolostron/backlog#19294](https://github.com/stolostron/backlog/issues/19294)
> 

Right. We already dynamically report deployment, connection and gateway status conditions on the `ManagedClusterAddon`. That actually makes me wonder - instead of splitting diagnostic info between the `ManagedClusterAddon` and this resource,  why don't we report all of it as status conditions on the `ManagedClusterAddon`?